api/src/main/java/io/opentelemetry/internal/StringUtils.java

```diff
@@ -57,5 +59,53 @@ public static boolean isValidMetricName(String metricName) {
     return metricName.matches(pattern);
   }
 
+  /**
+   * If given attribute is of type STRING and has more characters than given {@code limit} then
+   * return new AttributeValue with string truncated to {@code limit} characters.
+   *
+   * <p>If given attribute is of type STRING_ARRAY and non-empty then return new AttributeValue with
+   * every element truncated to {@code limit} characters.
+   *
+   * <p>Otherwise return given {@code value}
+   *
+   * @throws IllegalArgumentException if limit is zero or negative
+   */
+  public static AttributeValue cutIfNeeded(AttributeValue value, int limit) {
+    Utils.checkArgument(limit > 0, "attribute value limit cannot be %d", limit);
+
+    if (value == null
+        || (value.getType() != AttributeValue.Type.STRING
+            && value.getType() != AttributeValue.Type.STRING_ARRAY)) {
+      return value;
+    }
+
+    if (value.getType() == AttributeValue.Type.STRING_ARRAY) {
+      List<String> strings = value.getStringArrayValue();
+      if (strings.isEmpty()) {
+        return value;
+      }
+
+      String[] newStrings = new String[strings.size()];
+      for (int i = 0; i < strings.size(); i++) {
+        String string = strings.get(i);
+        newStrings[i] = string == null ? null : cutIfNeeded(string, limit);
+      }
+
+      return AttributeValue.arrayAttributeValue(newStrings);
+    }
+
+    String string = value.getStringValue();
```

Comment:
Believe it's possible for string to be null. Guess you can just use cutIfNeeded here too.

Reply:
I wanted to avoid allocating new `AttributeValue` unless needed...

Reply:
Well, actually it cannot. null value will not even reach this method. But I will add an `if` anyway

Reply:
Thanks, yeah didn't notice the reallocation of `AttributeValue` aspect. As far as null goes, I only thought of it from seeing the null check in this PR

https://github.com/open-telemetry/opentelemetry-java/pull/1437/files